tests/Unit/Evolution/DiscontinuousGalerkin/Test_ProjectToBoundary.cpp

```diff
@@ -0,0 +1,126 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/ApplyMatrices.hpp"
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Index.hpp"
+#include "DataStructures/SliceVariables.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Evolution/DiscontinuousGalerkin/ProjectToBoundary.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace {
+struct Var1 : db::SimpleTag {
+  using type = Scalar<DataVector>;
+};
+
+struct Var2 : db::SimpleTag {
+  using type = tnsr::I<DataVector, 2, Frame::Inertial>;
+};
+
+struct Var3 : db::SimpleTag {
+  using type = Scalar<DataVector>;
+};
+
+template <size_t Dim>
+Variables<tmpl::list<Var2, Var3>> polynomial_volume_data(
+    const tnsr::I<DataVector, Dim, Frame::Logical>& coords,
+    const Index<Dim>& powers) noexcept {
+  Variables<tmpl::list<Var2, Var3>> result(get<0>(coords).size(), 1.0);
+  for (size_t i = 0; i < Dim; ++i) {
+    get(get<Var3>(result)) *= pow(coords.get(i), powers[i]);
+    get<0>(get<Var2>(result)) *= 2.0 * pow(coords.get(i), powers[i]);
+    get<1>(get<Var2>(result)) *= 3.0 * pow(coords.get(i), powers[i]);
+  }
+  return result;
+}
+
+template <size_t Dim>
+void test(const Spectral::Quadrature quadrature) {
+  CAPTURE(Dim);
+  CAPTURE(quadrature);
+
+  MAKE_GENERATOR(gen);
+  UniformCustomDistribution<size_t> sdist{5, 10};
+
+  Mesh<Dim> volume_mesh{sdist(gen), Spectral::Basis::Legendre, quadrature};
+  Index<Dim> powers{};
+  for (size_t i = 0; i < Dim; ++i) {
+    powers[i] = volume_mesh.extents(i) - 2 - i;
+  }
+
+  const auto volume_data =
+      polynomial_volume_data(logical_coordinates(volume_mesh), powers);
+
+  for (const auto& direction : Direction<Dim>::all_directions()) {
+    const size_t sliced_dim = direction.dimension();
+    const size_t fixed_index = direction.side() == Side::Upper
+                                   ? volume_mesh.extents(sliced_dim) - 1
+                                   : 0;
+    const auto face_mesh = volume_mesh.slice_away(sliced_dim);
+    Variables<tmpl::list<Var2, Var3>> expected_face_values{};
+    if (quadrature == Spectral::Quadrature::GaussLobatto) {
+      expected_face_values = data_on_slice(volume_data, volume_mesh.extents(),
+                                           sliced_dim, fixed_index);
+    } else {
+      expected_face_values = polynomial_volume_data(
+          interface_logical_coordinates(face_mesh, direction), powers);
+    }
+    const Scalar<DataVector> expected_var1{face_mesh.number_of_grid_points(),
```

Comment:
[minor] Might be slightly clearer to call this `zeros` since you are using it in comparisons with Var1 and Var3. I don't feel strongly, up to you if you want to change it.

Reply:
I thought about this and the reason (which can reasonably be questioned) is I find the code easier to read when we have checks `a == expected_a` instead of `a == zero`. What do you think?

Reply:
The counter argument is `b == expected_a` :P. I don't know of a great solution here, so oh well, I'm happy for you to leave as is.

Reply:
haha self-rekt... Let me add a `expected_var3` :P